tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs

```diff
@@ -1696,4 +1696,12 @@ let ``Async.concat should work``() =
         |> AsyncSeq.ofSeq
         |> AsyncSeq.concatSeq
 
-      Assert.True(EQ expected actual)
\ No newline at end of file
+      Assert.True(EQ expected actual)
+
+#if (NETSTANDARD2_1 || NETCOREAPP3_0)
```

Comment:
Could you add a few more tests here please?  e.g. 

1. exceptions are raised
2. enumeration is not run until completion
3. cancellation tokens are used and cancellation requested

Reply:
I've added exception, cancellation and completion tests tests. Its somewhat tricky to test without a round-trip since F# does not yet have IAsyncEnumerable<_> .NET support. Can you clarify what you mean by "enumeration is not run until completion?" - I assumed that meant the workflow logic would not continue to run upon a "yield" until you called MoveNext()?